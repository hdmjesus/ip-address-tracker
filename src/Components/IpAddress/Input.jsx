import React from "react";
import iconArrrow from "../../Statics/images/icon-arrow.svg";
export const IpAddressInput = () => {
  return (
    <form className="  m-auto flex w-full ">
      <div className="flex min-w-200 justify-between bg-white rounded-lg  h-9 m-auto  w-full   ">
        <input
          className=" w-full p-2 text-xs bg-transparent outline-none "
          type="text"
          placeholder="Search for Any IP Address or Domain"
        />
        <button type="submit" className="bg-black h-9 w-9 rounded-r-lg">
          <img src={iconArrrow} alt="iconArrow" className="w-3 m-auto" />
        </button>
      </div>

      {/* <button type="submit" className="bg-black h-9 w-9">
        <img src={iconArrrow} alt="iconArrow" className="w-3" />
      </button> */}
    </form>
  );
};
