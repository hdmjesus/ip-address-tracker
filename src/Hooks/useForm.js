import { useState } from "react";

export const useForm = (initialState = {}) => {
  const [values, setValues] = useState(initialState);

  const reset = () => {
    setValues(initialState);
  };
  const handleInputChange = ({ target }) => {
    setValues({
      ...values,
      [target.name]: target.value,
    });
  };

  return [values, handleInputChange, reset];
};

// Este Hook nos permite obtener los valores y propiedades de un Input, ademas de tener los valores nuevos cuando este cambie.
