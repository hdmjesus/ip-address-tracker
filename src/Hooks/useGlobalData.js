import { useContext } from "react";
import { TextContext } from "../Context/StaticContext";

export function useGlobalData() {
  const data = useContext(TextContext);
  return data;
}
