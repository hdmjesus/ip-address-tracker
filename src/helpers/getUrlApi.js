import { apiKey, url } from "../config";

export const getUrlApi = (inputValue) => {
  let UrlApiIp;
  // Esta dondicion , devuelve el endpoint correspondiente a si es una direccion ip o un nombre de dominio
  if (
    inputValue.match(
      /^((25[0-5]|2[0-4]\d|[01]?\d\d?)\.){3}(25[0-5]|2[0-4]\d|[01]?\d\d?)$/
    )
  ) {
    UrlApiIp = `${url}?apiKey=${apiKey}&ipAddress=${inputValue}`;
    return UrlApiIp;
  } else {
    UrlApiIp = `${url}?apiKey=${apiKey}&domain=${inputValue}`;
    return UrlApiIp;
  }
};
