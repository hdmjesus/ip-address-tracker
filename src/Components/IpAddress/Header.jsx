import React from "react";
import { IpAddressInput } from "./Input";
import { IpAddressResultsContainer } from "./Results/Container";
import bgHeader from "../../Statics/images/pattern-bg.png";

export const IpAddressHeader = () => {
  return (
    <header
      className="w-full h-80 p-7  m-auto z-10"
      style={{
        backgroundImage: `url(${bgHeader})`,
        backgroundRepeat: "no-repeat",
      }}
    >
      <h1 className="  font-rubik font-bold text-white text-xl text-center mb-4">
        IP Address Tracker
      </h1>
      <IpAddressInput />
      <IpAddressResultsContainer />
    </header>
  );
};
