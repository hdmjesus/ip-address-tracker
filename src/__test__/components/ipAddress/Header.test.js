import React from "react";
import { mount, shallow } from "enzyme";
import { IpAddressHeader } from "../../../Components/IpAddress/Header";
import { TextContext } from "../../../Context/StaticContext";
import { useFetch } from "../../../Hooks/useFetch";
import { renderHook, act } from "@testing-library/react-hooks";
import { getUrlApi } from "../../../helpers/getUrlApi";
import bgImage from "../../../Statics/images/pattern-bg.png";

describe("Probando componente <IpAddressHeader/>", () => {
  const initialState = { data: null, loading: true, error: null };
  test("debe mostrar el componente sin la data", () => {
    const { result } = renderHook(() => useFetch(getUrlApi("www.hdjesus.com")));
    const { data, loading, error } = result.current;
    console.log(getUrlApi("www.hdjesus.com"));
    expect(data).toBe(null);
    expect(loading).toBe(true);
    expect(error).toBe(null);
  });

  test("debe mostrar el componente con la data", async () => {
    const { result, waitForNextUpdate } = renderHook(() =>
      useFetch(getUrlApi("8.8.8.8"))
    );
    await waitForNextUpdate();
    // {timeout:2000} se coloca para que el tiempo maximo para tener una respuesta de la API sea de 5000 ms

    const { data, loading, error } = result.current;

    expect(loading).toBe(false);
    expect(error).toBe(null);
  });

  test("debe mostrar el background imagen correcto", async () => {
    const { Provider } = TextContext;
    let loading = false;
    let data = null;
    const wrapper = mount(
      <Provider value={{ data, loading }}>
        <IpAddressHeader />
      </Provider>
    );

    expect(wrapper.find("header").prop("style").backgroundImage).toBe(
      `url(${bgImage})`
    );
    expect(wrapper).toMatchSnapshot();
  });
});
