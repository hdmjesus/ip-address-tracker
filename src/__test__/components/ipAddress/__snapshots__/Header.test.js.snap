// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente <IpAddressHeader/> debe mostrar el background imagen correcto 1`] = `
<header
  className="w-full h-80 p-7  m-auto z-10"
  style={
    Object {
      "backgroundImage": "url(pattern-bg.png)",
      "backgroundRepeat": "no-repeat",
    }
  }
>
  <h1
    className="  font-rubik font-bold text-white text-xl text-center mb-4"
  >
    IP Address Tracker
  </h1>
  <form
    className="  m-auto flex w-full "
    onSubmit={[Function]}
  >
    <div
      className="flex min-w-200 justify-between bg-white rounded-lg  h-9 m-auto  w-full   "
    >
      <input
        className=" w-full p-2 text-xs bg-transparent outline-none "
        name="ipInput"
        onChange={[Function]}
        placeholder="Search for Any IP Address or Domain"
        type="text"
        value=""
      />
      <button
        className="bg-black h-9 w-9 rounded-r-lg"
        type="submit"
      >
        <img
          alt="iconArrow"
          className="w-3 m-auto"
          src="icon-arrow.svg"
        />
      </button>
    </div>
  </form>
  <section
    className="w-11/12 bg-white p-3 rounded-md mt-5 m-auto flex flex-col"
  >
    <article
      className="m-auto mb-4 uppercase font-rubik "
    >
      <h4
        className="text-colorgray font-bold text-xs tracking-wider text-center"
      >
        IP Address
      </h4>
      <p
        className="font-medium text-xl text-blueBlack"
      >
         
      </p>
    </article>
    <article
      className="m-auto mb-4 uppercase font-rubik "
    >
      <h4
        className="text-colorgray font-bold text-xs tracking-wider text-center"
      >
        Location
      </h4>
      <p
        className="font-medium text-xl text-blueBlack"
      >
         
        , 
      </p>
    </article>
    <article
      className="m-auto mb-4 uppercase font-rubik "
    >
      <h4
        className="text-colorgray font-bold text-xs tracking-wider text-center"
      >
        ISP
      </h4>
      <p
        className="font-medium text-xl text-blueBlack"
      >
         
      </p>
    </article>
  </section>
</header>
`;
